plugins/datadog-entity-sync-node/src/transforms/defaultEntitySerializer.ts

```diff
@@ -0,0 +1,199 @@
+  entity: ComponentEntity | ApiEntity | SystemEntity | ResourceEntity,
+  repoContext?: CodeRepositoryContext,
+  extraInfo?: ExtraSerializationInfo,
+) {
+  const hasTechDocs = Boolean(
+    entity.metadata.annotations?.[TECHDOCS_ANNOTATION],
+  );
+
+  for (const { title, type, url } of entity.metadata.links ?? []) {
+    if (title && url) {
+      yield {
+        name: title,
+        type: type && LINK_TYPES.includes(type) ? type : 'other',
+        url: url,
+      };
+    }
+  }
+
+  if (extraInfo?.appBaseUrl) {
+    const ref = getCompoundEntityRef(entity);
+    yield {
+      name: 'Backstage',
+      type: 'doc',
+      provider: 'backstage',
+      url: `${extraInfo.appBaseUrl}/catalog/${ref.namespace}/${ref.kind}/${ref.name}`,
+    };
+
+    if (hasTechDocs) {
+      yield {
+        name: 'TechDocs',
+        type: 'doc',
+        provider: 'backstage',
+        url: `${extraInfo.appBaseUrl}/docs/${ref.namespace}/${ref.kind}/${ref.name}`,
+      };
+    }
+  }
+
+  if (repoContext) {
+    yield {
+      name: 'Source',
+      type: 'repo',
+      provider: repoContext.provider,
+      url: repoContext.url,
+    };
+  }
+}
+
+/**
+ * Default serializer that preserves the original Backstage entity structure
+ * while adding enrichments like resolved owner, combined tags, and auto-generated links.
+ *
+ * @throws Error if the entity is not a component, api, system, or resource.
+ *
+ * @param entity - The Backstage Catalog Entity to serialize (should be a component entity).
+ * @param extraInfo - Optional context for serialization.
+ * @returns Enhanced Backstage entity with enrichments for Datadog.
+ */
+export function defaultEntitySerializer(
+  entity: Entity | ComponentEntity | ApiEntity | SystemEntity | ResourceEntity,
+  extraInfo?: ExtraSerializationInfo,
+) {
+  ensureComponent(entity);
+
+  const { metadata, spec } = entity;
+  const repoContext = resolveRepositoryInfo(entity);
+  const entityOwnerRef = getEntityRelationRefs(entity, RELATION_OWNED_BY).at(0);
+
+  return {
+    ...entity, // Keep original entity structure
+    metadata: {
+      ...metadata,
+      name:
+        metadata.annotations?.['datadoghq.com/service-name'] ?? metadata.name,
+      ...valueGuard(metadata.description, description => ({
+        description,
+      })),
+      ...valueGuard(entityOwnerRef, ownerRef => ({
+        owner: parseEntityRef(ownerRef.targetRef).name,
+      })),
+      tags: [...labelsToTags(entity), ...(metadata.tags ?? [])],
+      ...valueGuard(
+        Array.from(getDatadogStyleLinks(entity, repoContext, extraInfo)),
+        links => ({
+          links,
+        }),
+      ),
+    },
+    spec: {
+      ...spec,
+    },
+    ...valueGuard(repoContext, repo => ({
+      datadog: {
+        codeLocations: [
+          {
+            repositoryURL: repo.repositoryURL,
+            ...valueGuard(repo.path, path => ({
+              paths: [path],
+            })),
+          },
+        ],
+      },
+    })),
```

Comment:
Should this be included in the spec? 🤔 I would say that would seem odd to me to be in the backstage entity spec. But maybe this is a more just of how datadog interprets it.

Reply:
saw this was changed.